### PR TITLE
Update help description for make.sh

### PR DESCRIPTION
**Pull request by lzutao**

### CODE REVIEW

File: make.sh [hunk 1]
```diff
@@ -136,6 +157,6 @@ case "$1" in
   "linux32" ) build_linux32;;
   "msvc_update_genfiles" ) msvc_update_genfiles;;
   * )
-    echo "Usage: $0 ["`grep '^  "' $0 | cut -d '"' -f 2 | tr "\\n" "|"`"]"
```
Comment (aquynh) [inline make.sh]: the old solution is nice because it is quite future proof, so i feel that we better keep that approach?

Comment (lzutao) [reply make.sh]: I think keeping description for each build option in `usage` is quite easy.
If this PR is merged, everyone would take a look in `usage` function first, then 
edit `usage` description appropriate.

File: make.sh [hunk 2]
```diff
@@ -3,21 +3,42 @@
...
+    cat 1>&2 <<EOF
+make.sh - The build script for unicorn engine
+USAGE:
+    $ ./make.sh [OPTIONS]
+OPTIONS:
+                            Build the project
+    asan                    Build for ASan
+    install                 Install the project
+    uninstall               Uninstall the project
+    macos-universal         Build universal binaries on macOS
```
Comment (lzutao) [inline make.sh]: I don't know if this description is correct or not.

File: make.sh [hunk 3]
```diff
@@ -136,6 +157,6 @@ case "$1" in
   "linux32" ) build_linux32;;
   "msvc_update_genfiles" ) msvc_update_genfiles;;
```
Comment (lzutao) [inline make.sh]: This one I have no idea about its description. Could you give a suggestion?

Comment (aquynh) [reply make.sh]: this is to support MSVC build, which cannot automate this copying step itself.

File: make.sh (hunk 2 above)
Comment (aquynh) [reply make.sh]: yes this is correct

File: make.sh [hunk 4]
```diff
@@ -3,21 +3,42 @@
...
-  UNICORN_ASAN=yes
-  ${MAKE} V=1
+  env UNICORN_DEBUG=yes UNICORN_ASAN=yes "${MAKE}" V=1
 }
 
 # build iOS lib for all iDevices, or only specific device
 build_iOS() {
-  IOS_SDK=`xcrun --sdk iphoneos --show-sdk-path`
-  IOS_CC=`xcrun --sdk iphoneos -f clang`
+  IOS_SDK=$(xcrun --sdk iphoneos --show-sdk-path)
```
Comment (aquynh) [inline make.sh]: @radare what do you think, i am unsure if this is a good fix, that is portable across all shells?

**Comment by lzutao** [comment]
Also a little bike-shed question: What keeps us from using python3 to build unicorn instead of python2?
Is it QEMU or LLVM part? If not, do you have a wish to convert to python3?

File: make.sh (hunk 1 above)
Comment (aquynh) [reply make.sh]: agreed, let see if there is any other comments on Shell coding.

File: make.sh (hunk 4 above)
Comment (lzutao) [reply make.sh]: I think it is portable. http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_03

**Comment by aquynh** [comment]
Qemu part. You are welcome to send in PR to support Python3!

File: make.sh (hunk 4 above)
Comment (aquynh) [reply make.sh]: bash? but we do not depend on a specific shell, so please make sure this works with other non-bash shells too.

Comment (lzutao) [reply make.sh]: I think all POSIX shell should follow [the opengroup](https://en.wikipedia.org/wiki/The_Open_Group) specification above.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**